### DISCUSSION TITLE
Add environment variables to override directories for buildtrees, installed, packages

### DISCUSSION THREAD

**Discussion by ned14** [microsoft/vcpkg#19147] [Ideas]
**Is your feature request related to a problem? Please describe.**

We are trialling vcpkg as a git submodule to replace cmake hunter. The source code lives on a Samba share, and is used by multiple build environments including Windows and Linux. We build by creating a local build directory on the target machine, and for cmake to use the source tree on the Samba share.

Right now, vcpkg appears to always write into `VCPKG_ROOT` which is problematic for our use case, not least that writing build over a network drive is very slow (SMB is fast to read unmodified files, they get oplock cached locally, but it's dog slow if anything ever gets modified).

**Proposed solution**

https://github.com/microsoft/vcpkg/blob/master/docs/users/config-environment.md has a `VCPKG_DOWNLOADS` environment variable to override the downloads. I'd like `VCPKG_BUILDTREES`, `VCPKG_INSTALLED` and `VCPKG_PACKAGES` environment variables too so `VCPKG_ROOT` no longer is modified over a network drive. Then I can tell vcpkg to use directories inside my local build directory, and everything goes very considerably faster. Plus, the same `VCPKG_ROOT` can then be shared between Linux and Windows.

**Describe alternatives you've considered**

None. Single source many target architectures is a hard requirement here. If a package manager can't do that, it's a showstopper.

**Comment by PhoebeHui** [comment]
I don't have any good suggestions for this, @strega-nil-ms, any ideas?

**Comment by strega-nil-ms** [comment]
I believe we already have these as experimental flags - `x-install-root`, `x-packages-root`, and `x-buildtrees-root`; it would be pretty trivial to also add environment variables for these, and @ned14 that might solve your problem?

**Comment by 0xjairo** [comment]
Issue https://github.com/microsoft/vcpkg/issues/16098 mentions the experimental command line flags but I'm unclear if that's something we can use in manifest mode within CMake

```
  --x-buildtrees-root=<path>      (Experimental) Specify the buildtrees root directory
  --x-install-root=<path>         (Experimental) Specify the install root directory
  --x-packages-root=<path>        (Experimental) Specify the packages root directory
```

**Reply by dg0yt** [reply]
You can pass specify additional vcpkg parameters in CMake variable `VCPKG_INSTALL_OPTIONS:
https://github.com/microsoft/vcpkg/blob/master/docs/users/manifests.md#vcpkg_install_options

**Reply by 0xjairo** [reply]
this worked!, thanks! In my case I did this:
```sh
 cmake -B build -S . -G Ninja  -DVCPKG_INSTALL_OPTIONS="--x-buildtrees-root=build/buildtrees"
```
